test/tcp/TcpClient.cpp

```diff
@@ -27,6 +28,46 @@ class Callback : public AsyncSocket::ConnectCallback {
     std::cout << "TODO error" << ex.what() << " " << ex.getType() << "\n";
   }
 };
+
+class FollyKeepaliveTimer : public KeepaliveTimer {
+ public:
+  explicit FollyKeepaliveTimer(folly::EventBase& eventBase)
+      : eventBase_(eventBase) {
+    running_ = std::make_shared<bool>(false);
+  }
+
+  std::chrono::milliseconds keepaliveTime() override {
+    return std::chrono::milliseconds(5000);
+  };
+
+  void schedule() {
```

Comment:
can you add a comment somewhere that this method as well as close method are supposed to be executed on the same thread so we don't have race condition around the value of running?
Maybe a general comment for the class in the header file would be enough..